HW29/src/main/java/org/example/models/Cart.java

```diff
@@ -0,0 +1,45 @@
+package org.example.models;
+
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+@Scope("prototype")
+public class Cart {
+
+    private List<Product> productsInCart = new ArrayList<>();
+
+    AnnotationConfigApplicationContext context =
```

Comment:
Spring context, in very very simple terms, is a holder for objects that could be used in spring applications. So, ideally in the scope of the application must be only one context (or there should be a strong reason to have more than one context).

Here you create a new context and a bit later take a repository bean from it. This means that the context that you declared in the main method does not have this bean (because this is two different contexts).

If you need a bean use `@Autowired` annotation, like this:
```
@Autowired
private ProductRepository product repository;
```
or inject `ProductRepository` into `Cart` during `Cart` creation in the `Configuration` class, something like it:
Cart class:
```
public class Cart{
  private ProductRepository productRepository;

  public Cart(ProductRepository productRepository){
    this.productRepository = productRepository;
  }
...
```

Configuration class:
```
...
@Bean
public Cart cart(ProductRepository productRepository){
  return new Cart(productRepository);
}
...
```